src/main/java/org/generationcp/breeding/manager/listmanager/util/GermplasmListExporter.java

```diff
@@ -421,33 +423,126 @@ public void exportGermplasmListCSV(final String fileName, final Table listDataTa
+		int colIndex) {
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.PREFERRED_ID))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.PREFERRED_ID))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.PREFERRED_ID.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.PREFERRED_NAME))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.PREFERRED_NAME))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.PREFERRED_NAME.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.GERMPLASM_DATE))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.GERMPLASM_DATE))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.GERMPLASM_DATE.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.GERMPLASM_LOCATION))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.GERMPLASM_LOCATION))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.GERMPLASM_LOCATION.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_NAME))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_NAME))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.BREEDING_METHOD_NAME.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_ABBREVIATION))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_ABBREVIATION))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.BREEDING_METHOD_ABBREVIATION.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_NUMBER))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_NUMBER))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.BREEDING_METHOD_NUMBER.getName(), true));
+		}
+
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_GROUP))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.BREEDING_METHOD_GROUP))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.BREEDING_METHOD_GROUP.getName(), true));
+		}
+
+		/**
+		 * TODO verify explicit exclusion in {@link GermplasmListExporter#getVisibleColumnMap(com.vaadin.ui.Table)}
+		 */
+		if (visibleColumnMap.containsKey(this.getColumnNamesTermId(ColumnLabels.FGID))
+			&& visibleColumnMap.get(this.getColumnNamesTermId(ColumnLabels.FGID))) {
+			exportColumnHeaders.add(new ExportColumnHeader(colIndex++, ColumnLabels.FGID.getName(), true));
+		}
```

Comment:
@gelli-leafnode @darla-leafnode 
In https://github.com/IntegratedBreedingPlatform/BreedingManager/commit/5b54ebef8534d17a933359bf2f322c4c26b3a327 (https://github.com/IntegratedBreedingPlatform/BreedingManager/pull/321) `FGID` and `MGID` have been explicitly excluded from the `visibleColumnMap`, which is used for the export. **Why is that?** 
In this branch I have included the _added columns_ as part of the export, but it won't show those two because of the exclusion.

Reply:
Hi @nahuel-soldevilla , 

According to issue fixed in IBP-2796: fgid/mgid should only be included in export if from crossing manager [`IBP-2796`](https://ibplatform.atlassian.net/browse/IBP-2796?focusedCommentId=41029)

//FGID and MGID are addable columns in BreedingManager but standard in CrossList.
//We exclude them so as not to cause conflict in GermplasmExportedWorkbook code handling for these columns

Reply:
thanks @gelli-leafnode 
Yes I saw your comment in the code, but I found it strange given that the columns FGID and MGID are visible in the table, I don't see why they shouldn't be there in the export file.
So if understand correctly, **this was not a requirement from IBP**, but rather a way to prevent that bug when you try to export list that are coming from a cross, and in fact I can verify that if I remove the exclusion, it seems the the code to export these columns was always there but somehow something change in middle and it got broken. 
In my opinion this is because of the the data available from the export is not being populated in the same way that the data in the table. If that the case I'll try fix the issue or at least create an improvement.
**But if this was a requirement from IBP, please let me know.** No rush, I know it's a holiday there.

cc @darla-leafnode @clarysabel

Reply:
@darla-leafnode @gelli-leafnode I just merged this ticket today. Please let us know if you have any concern and we can follow up on it